src/javascripts/Services/recordViewService.js

```diff
@@ -331,5 +331,11 @@ ngapp.service('recordViewService', function($timeout, layoutService, settingsSer
             scope.view.linkTo(treeView);
             treeView.linkTo(scope.view);
         };
+
+        scope.syncWithLinkedViews = function(record) {
```

Comment:
This code looks like it's here to sync the record view with the referenced by view, but it's kind of messy.  Traversing through the tree view to the tree view's linked record view.  We shouldn't do it this way.  If we want to link the record view to the referenced by view we should link the referenced by view *directly* to the record view.